app/src/main/java/ch/epfl/sweng/rps/models/Game.kt

```diff
@@ -15,8 +15,8 @@ import com.google.firebase.Timestamp
 data class Game(
     val id: String,
     val players: List<String>,
-    val rounds: Map<String, Round>,
-    val current_round: Int,
+    val rounds: MutableMap<String, Round>,
+    var current_round: Int,
```

Comment:
You shouldn't modifiy this class. `MutableMap` extends `Map`, so there is no need to change it to `MutableMap`. For the round, you should use `game.copy()` to have a new instance with your new `current_round`.